### PR TITLE
Feature create

### PR DESCRIPTION
**Pull request by AlexP953**
Added VC of the create method.
Checked with changes to avoid conflicts

### CODE REVIEW

File: app/controllers/CreateController.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+class CreateController extends ApplicationController
+{
+	public function createAction()
+	{
+			if ($_SERVER['REQUEST_METHOD'] === 'POST') {
```
Comment (Kaylen7) [inline app/controllers/CreateController.php]: Just a suggestion. Controller class (which is parent of ApplicationController) has public method `isPost` that returns boolean check. You could implement it here instead of `$_SERVER['REQUEST_METHOD'] === 'POST'`. It's basically the same code. 

I don't really have any explanation why `if($this->isPost())` would be better than your option, so it really is just a suggestion. 🤷‍♀️

File: app/models/Database.php [hunk 2]
```diff
@@ -88,7 +88,37 @@ private function updateInnerTasks(string $id, string $parameter, string $input):
...
+            }
+        } while ($taskExists);
+    
+        array_push($this->tasks, $newTask);
+        $this->updateDatabase();
+        return $this->tasks;
+    }
+
     private function updateDatabase(){
+        $this->tasks = array_values($this->tasks);
```
Comment (Kaylen7) [inline app/models/Database.php:121]: Why this change?

**Review by Kaylen7 — CHANGES_REQUESTED** [review]
There's a couple of things that might need review, but the rest is ok.

File: app/controllers/CreateController.php (hunk 1 above)
Comment (AlexP953) [reply app/controllers/CreateController.php]: changed $_SERVER['REQUEST_METHOD'] === 'POST' to if($this->isPost()). Thanks!

File: app/models/Database.php (hunk 2 above)
Comment (AlexP953) [reply app/models/Database.php:121]: Because without this line the JSON was out of order:

Without code
ex  without code

With code
ex  with code

Comment (Kaylen7) [reply app/models/Database.php:121]: But there's no need to order it, no? I thought our db was: 
Screenshot 2024-11-21 at 17 05 58

Comment (Kaylen7) [reply app/models/Database.php:121]: I mean, when I test it with editTask it gets correctly updated with and without that line... 🧐 

Maybe we can keep updateDatabase() method simple (it only handles copying "tasks" object into JSON file) and then formatting $this->tasks can be done in each method? 

In what method do you find these changes? Maybe I can test it out too

Comment (AlexP953) [reply app/models/Database.php:121]: It only happens when deleting a task. From what I saw, in PHP if you convert an array to JSON, if it doesn't have consecutive numeric indexes, it becomes a JSON object and changes the format, so it adds the indexes "1":{}. That's why I reindex the array with array_values, so that when rewriting the JSON the format isn't changed. It doesn't affect much in principle but it would constantly be changing the JSON format that we had agreed upon at the beginning.

Since it only happens with DELETE, I can pass it to its corresponding method as well.

Comment (Kaylen7) [reply app/models/Database.php:121]: Oh, I see. Nono, then it's ok that it is here. Thanks for explaining :)

File: web/javascripts/functions.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  const setModalContent = () => {
+    modalTitle.textContent = title;
+    modalBody.textContent = message;
+    closeBtn.textContent = buttonText;
+  };
+
+  const closeModal = () => {
+    modal.classList.add('hidden');
+    if(redirect) window.location.href='/';
```
Comment (Kaylen7) [inline web/javascripts/functions.js]: Maybe it would be interesting to leave redirection open, so that when modal is shown in `/create` it redirects to `view/:id`, but when it is shown when triggered by `/delete` (we're yet to see how to add multiple ids there 😅) it redirects to `/`. 

@AlexP953, what do you think?

File: app/views/scripts/create/create.phtml
Comment (Kaylen7) [inline app/views/scripts/create/create.phtml:1]: I think description and tags should not be required, as maybe there's tasks that don't necessarily have them.

Remember to change this ruleset in `CreateController->createAction()` as it now enforces to fill in all fields as well.

File: app/controllers/CreateController.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+					$description = $_POST['description'] ?? null;
+					$tags = $_POST['tags'] ?? null;
+					$start_time = $_POST['start_time'] ?? null;
+					$end_time = $_POST['end_time'] ?? null;
+					$created_by = $_POST['created_by'] ?? null;
+					$state = $_POST['state'] ?? null;
+	
+					// Validación de campos
+					if (!$title || !$description || !$tags || !$start_time || !$end_time || !$created_by || !$state) {
+							echo "Error: Faltan datos.";
```
Comment (Kaylen7) [inline app/controllers/CreateController.php]: I leave here a reminder to manage errors. Here this line prints it out in the frontend, we should probably hide it or consider throwing redirection towards an error page.

File: web/javascripts/functions.js (hunk 3 above)
Comment (AlexP953) [reply web/javascripts/functions.js]: Sounds good! I've changed the code so that the new route can be sent by parameter. I want to look into how to redirect to view/:id but for now it can stay like this to continue moving forward. Thanks!

File: app/views/scripts/create/create.phtml
Comment (AlexP953) [reply app/views/scripts/create/create.phtml:1]: You're right. I've changed it!

File: app/controllers/CreateController.php (hunk 4 above)
Comment (AlexP953) [reply app/controllers/CreateController.php]: I've changed it!
